listener/upload_listener.py

```diff
@@ -90,11 +94,26 @@ def process_upload(msg):
         # Download archive an parse
         with tempfile.NamedTemporaryFile(delete=True) as tmp_file:
             LOGGER.debug("Writing temp file to %s", tmp_file.name)
-            response = session.get(upload_data["url"], stream=True, allow_redirects=True)
-            for chunk in response.iter_content(chunk_size=1024):
-                if chunk:  # filter out keep-alive new chunks
-                    tmp_file.write(chunk)
-            tmp_file.flush()
+            # Grab the uploaded archive and write it locally so we can dig Important THings out of it
+            # If the read-attempt fails, retry up to MAX_ARCHIVE_RETRIES times before logging the failure and moving on
+            for tries in range(MAX_ARCHIVE_RETRIES):
+                try:
+                    response = session.get(upload_data["url"], stream=True, allow_redirects=True)
+                    for chunk in response.iter_content(chunk_size=1024):
+                        if chunk:  # filter out keep-alive new chunks
+                            tmp_file.write(chunk)
+                    tmp_file.flush()
+                    break
+                except Exception: # pylint: disable=broad-except
+                    # on last try, log and fail this upload. Else, log try again
+                    if (tries+1) == MAX_ARCHIVE_RETRIES:
+                        ARCHIVE_RETRIEVE_FAILURE.inc()
+                        LOGGER.exception("Unable to retrieve archive")
```

Comment:
LOGGER.exception appends the exception as one line after the text, it would be better with space, for example:

`LOGGER.exception("Unable to retrieve archive: ")`